src/eos-dropbox-app

```diff
@@ -183,35 +187,39 @@ class DropboxLauncher():
 
         try:
-            bus = Gio.bus_get_sync(Gio.BusType.SESSION, None)
-            proxy = Gio.DBusProxy.new_sync(bus, Gio.DBusProxyFlags.NONE,
-                                           None,
-                                           'org.freedesktop.portal.Desktop',
-                                           '/org/freedesktop/portal/desktop',
-                                           'org.freedesktop.portal.OpenURI',
-                                           None)
+            self._bus = Gio.bus_get_sync(Gio.BusType.SESSION, None)
+            self._proxy = Gio.DBusProxy.new_sync(self._bus, Gio.DBusProxyFlags.NONE,
+                                                 None,
+                                                 'org.freedesktop.portal.Desktop',
+                                                 '/org/freedesktop/portal/desktop',
+                                                 'org.freedesktop.portal.OpenURI',
+                                                 None)
 
             logging.info("Opening Dropbox directory at {}...".format(path))
-            handle = proxy.call_with_unix_fd_list('OpenFile',
-                                                  GLib.Variant('(sha{sv})',
-                                                               ('', 0, GLib.Variant('a{sv}', None))),
-                                                  Gio.DBusCallFlags.NONE,
-                                                  -1,
-                                                  Gio.UnixFDList.new_from_array([fd]))
-
-            # The OpenURI method returns a handle to a 'request object', which stays
-            # alive for the duration of the user interaction related to the method call,
-            # so we connect to its Response signal to know when it's all over.
-            bus.signal_subscribe('org.freedesktop.portal.Desktop',
-                                 'org.freedesktop.portal.Request',
-                                 'Response',
-                                 handle,
-                                 None,
-                                 Gio.DBusSignalFlags.NO_MATCH_RULE,
-                                 self._responseReceived)
-
-        except GLib.Error as e:
-            self._exitOnError("Could not launch the OpenURI portal for {}: {}".format(uri, e.message))
+            [result, _] = self._proxy.call_with_unix_fd_list_sync('OpenFile',
+                                                                  GLib.Variant('(sha{sv})',
+                                                                               ('', 0, None)),
+                                                                  Gio.DBusCallFlags.NONE,
+                                                                  -1,
+                                                                  Gio.UnixFDList.new_from_array([fd]),
+                                                                  None)
+            handle = result.unpack()[0];
```

Comment:
Here I'd rather see:

```python
handle, = result.unpack()
```

Because you're expecting the result to be a one-element tuple. If you prefer `[0]` then there should definitely not be a trailing `;`.

Reply:
Ok.